internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -515,7 +518,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// We pass a deepcopy because the update method doesn't update status,
 	// but calling update resets any pending status changes.
 	updated := cr.DeepCopyObject().(client.Object)
-	if err := r.client.Update(ctx, updated); err != nil {
+	if err := r.client.Apply(ctx, updated, mergeOptions(filterToXRPatches(tas))...); err != nil {
```

Comment:
What are the implications of changing from `Update` to `Apply` (i.e. patch) here?

Reply:
Added the following comments for clarification:
```
	// call Apply so that we do not just replace fields on existing XR but
	// merge fields for which a merge configuration has been specified.
	// For fields for which a merge configuration does not exist, the
	// behavior will be a replace from updated.
```
The idea is we now have control over whether fields of the XR get replaced or merged according to the rules specified on a patch targeting them (XR-targeting patches, if any).

Reply:
Thank you!

I think there is a small behaviour change here in that we will now patch rather than update, regardless of the supplied merge options. That is, if an XR field was unset we'd previously persist that but now won't. That said, I don't think this will matter in practice because what we're doing here is persisting the result of the above `r.composite.Render` calls, and I believe the patches applied by those calls are purely additive (i.e. they never unset fields).